app/page.tsx

```diff
@@ -3,72 +3,80 @@
 import { useAuth } from "@/services/AuthContext";
 import TopBar from "@/components/TopBar";
 import BottomNavigationBar from "@/components/BottomNavigationBar";
-import { getSessionOccurrences, getSessions } from "@/api/afdb/session";
+import { getSessions, getGroupUser, getGroupSessions, getGroupTypes } from "@/api/afdb/session";
 import { useState, useEffect } from "react";
-import { LiveClasses } from "./types";
+import { GroupUser, GroupSession, Session } from "./types";
 import Link from "next/link";
 import PrimaryButton from "@/components/Button";
 import Loading from "./loading";
-import { isSameDay, formatCurrentTime, formatSessionTime } from "@/utils/dateUtils";
+import { formatCurrentTime, formatSessionTime } from "@/utils/dateUtils";
 import { generateQuizLink } from "@/utils/quizUtils";
 
 export default function Home() {
-  const { loggedIn, userId } = useAuth();
-  const [liveClasses, setLiveClasses] = useState<LiveClasses[]>([]);
+  const { loggedIn, userId, userDbId } = useAuth();
+  const [liveClasses, setLiveClasses] = useState<Session[]>([]);
   const [isLoading, setIsLoading] = useState(true);
-  const [quizzes, setQuizzes] = useState<LiveClasses[]>([]);
+  const [quizzes, setQuizzes] = useState<Session[]>([]);
   const commonTextClass = "text-gray-700 text-sm md:text-base mx-6 md:mx-8";
   const infoMessageClass = "flex items-center justify-center text-center h-72 mx-4 pb-40";
 
-  const fetchSessionOccurrencesAndDetails = async () => {
+  const fetchUserSessions = async () => {
     try {
-      const sessionOccurrenceData = await getSessionOccurrences();
-      const currentDate = new Date();
-      currentDate.setHours(0, 0, 0, 0);
-
-      const todaySessions = [];
-
-      for (const sessionOccurrence of sessionOccurrenceData) {
-        const sessionStartTime = new Date(sessionOccurrence.start_time);
-        const formattedSessionTime = formatSessionTime(sessionOccurrence.end_time);
-        const formattedCurrentTime = formatCurrentTime(new Date().toISOString());
-
-        if (isSameDay(sessionStartTime, currentDate) && formattedSessionTime > formattedCurrentTime) {
-          try {
-            const sessionDetail = await getSessions(sessionOccurrence.session_fk);
-            todaySessions.push({
-              sessionOccurrence,
-              sessionDetail,
-            });
-          } catch (error) {
-            console.error("Error fetching session details for session ID:", sessionOccurrence.session_fk, error);
-          }
+      const currentDay = new Date().getDay();
+      const groupUserData = await getGroupUser(userDbId!);
+
+      const groupSessions = await Promise.all(groupUserData.map(async (userData: GroupUser) => {
+        const groupType = await getGroupTypes(userData.group_type_id);
+
+        const groupTypeIds = groupType.map((type: any) => type.id);
```

Comment:
We should make an API on DB service for this call, so that instead of making 3 calls, we can make one call and DB service can handle the computation. But later... Can you bring this up when we meet?

Reply:
Yeah…even I was thinking about that